tests/test_fitzhughnagumo.py

```diff
@@ -0,0 +1,164 @@
+#
+# test_mixed_integrator_numeric.py
+#
+# This file is part of the NEST ODE toolbox.
+#
+# Copyright (C) 2017 The NEST Initiative
+#
+# The NEST ODE toolbox is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation, either version 2 of
+# the License, or (at your option) any later version.
+#
+# The NEST ODE toolbox is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+# of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+
+import json
+import os
+import pytest
+import unittest
+import sympy
+import numpy as np
+from scipy.signal import find_peaks
+from sympy.parsing.sympy_parser import parse_expr
+from sympy.solvers import solve
+from sympy import Symbol
+
+
+try:
+    import matplotlib as mpl
+    mpl.use('Agg')
+    import matplotlib.pyplot as plt
+    INTEGRATION_TEST_DEBUG_PLOTS = True
+except:
+    INTEGRATION_TEST_DEBUG_PLOTS = False
+
+
+import odetoolbox
+from odetoolbox.mixed_integrator import MixedIntegrator
+
+from math import e
+from sympy import exp, sympify
+import sympy.parsing.sympy_parser
+
+import scipy
+import scipy.special
+import scipy.linalg
+
+try:
+    import pygsl.odeiv as odeiv
+    PYGSL_AVAILABLE = True
+except ImportError as ie:
+    PYGSL_AVAILABLE = False
+
+
+def open_json(fname):
+    absfname = os.path.join(os.path.abspath(os.path.dirname(__file__)), fname)
+    with open(absfname) as infile:
+        indict = json.load(infile)
+    return indict
+
+
+class TestFitxhughNagumo(unittest.TestCase):
+    
+    """
+    Implementing the fitzhughNagumo model starting from equilibrium values, and performing a test that if the external current crosses a certain threshold value, regular spikes are obtained.
+    This function tests if the number of spikes cross 20 in that case. 
+    Additionally, plots of V and W vs time are obtained for different values of current, and a FI curve is also plotted. 
+    """
+    
+    def initial__values(self, curr):
+       I_ext = Symbol("I_ext")
+       V = Symbol("V") 
+       expr = solve((sympy.parsing.sympy_parser.parse_expr("8*V**3 + 6*V + 21 - 24*I_ext")), V) # expr gives a list of three roots for V: first two are complex, third one is real
+       final_val_V = (expr[2].subs(I_ext,curr)).evalf()
+       final_val_W = ((10*final_val_V) + 7)/8
+       return float(final_val_V), float(final_val_W)
+
+    @pytest.mark.skipif(not PYGSL_AVAILABLE, reason="Need GSL integrator to perform test")
+    def test_fitzhugh_nagumo(self):
+        debug = True
+
+        h = 1    # [ms] #time steps
+        T = 1000  # [ms] #total simulation time 
+        n = 25 #total number of current values between 0 and 1
```

Comment:
I would suggest to drastically lower this number if we're not making any plots, just so the testsuite doesn't take too long to complete on Travis. Possibly even a manually curated list (``I_ext = np.array([0., ..., 1.])``)